net/core/dev.c

```diff
@@ -10506,6 +10565,19 @@ static int __init net_dev_init(void)
 		init_gro_hash(&sd->backlog);
 		sd->backlog.poll = process_backlog;
 		sd->backlog.weight = weight_p;
+#ifdef CONFIG_XDP_LUA
+		lw->L = luaL_newstate();
+		WARN_ON(!lw->L);
+
+		if (!lw->L)
```

Comment:
We need to improve this, @lneto any suggestions?

Reply:
IMHO, we should fail XDPLua initiation completely and destroy all states we have created so far.. and, of course, disable XDPLua execution in runtime putting a guard..

Reply:
btw, another example of what I was arguing on slack is to move all this code to a `xdplua_init()` function